plugins/orders.cpp

```diff
@@ -0,0 +1,788 @@
+    }
+}
+
+template<typename B>
+static void json_array_to_bitfield(B & bits, Json::Value & arr)
+{
+    if (arr.size() == 0)
+    {
+        return;
+    }
+
+    for (Json::ArrayIndex i = arr.size() - 1; i != 0; i--)
+    {
+        int current;
+        if (get_bitfield_field(&current, bits, arr[i].asString()))
+        {
+            if (!current && set_bitfield_field(&bits, arr[i].asString(), 1))
+            {
+                Json::Value removed;
+                arr.removeIndex(i, &removed);
+            }
+        }
+    }
+}
+
+template<typename D>
+static df::itemdef *get_itemdef(int16_t subtype)
+{
+    return D::find(subtype);
+}
+
+template<typename D>
+static df::itemdef *get_itemdef(const std::string & subtype)
+{
+    for (auto it = D::get_vector().begin(); it != D::get_vector().end(); it++)
+    {
+        if ((*it)->id == subtype)
+        {
+            return *it;
+        }
+    }
+    return nullptr;
+}
+
+template<typename ST>
+static df::itemdef *get_itemdef(color_ostream & out, df::item_type type, ST subtype)
+{
+    switch (type)
+    {
+    case item_type::AMMO:
+        return get_itemdef<df::itemdef_ammost>(subtype);
+    case item_type::ARMOR:
+        return get_itemdef<df::itemdef_armorst>(subtype);
+    case item_type::FOOD:
+        return get_itemdef<df::itemdef_foodst>(subtype);
+    case item_type::GLOVES:
+        return get_itemdef<df::itemdef_glovesst>(subtype);
+    case item_type::HELM:
+        return get_itemdef<df::itemdef_helmst>(subtype);
+    case item_type::INSTRUMENT:
+        return get_itemdef<df::itemdef_instrumentst>(subtype);
+    case item_type::PANTS:
+        return get_itemdef<df::itemdef_pantsst>(subtype);
+    case item_type::SHIELD:
+        return get_itemdef<df::itemdef_shieldst>(subtype);
+    case item_type::SHOES:
+        return get_itemdef<df::itemdef_shoesst>(subtype);
+    case item_type::SIEGEAMMO:
+        return get_itemdef<df::itemdef_siegeammost>(subtype);
+    case item_type::TOOL:
+        return get_itemdef<df::itemdef_toolst>(subtype);
+    case item_type::TOY:
+        return get_itemdef<df::itemdef_toyst>(subtype);
+    case item_type::TRAPCOMP:
+        return get_itemdef<df::itemdef_trapcompst>(subtype);
+    case item_type::WEAPON:
+        return get_itemdef<df::itemdef_weaponst>(subtype);
+    default:
+        out << COLOR_YELLOW << "Unhandled raw item type in manager order: " << enum_item_key(type) << "! Please report this bug to DFHack." << std::endl;
+        return nullptr;
+    }
+}
+
+static command_result orders_export_command(color_ostream & out, const std::string & name)
+{
+    if (!is_safe_filename(out, name))
+    {
+        return CR_WRONG_USAGE;
+    }
+
+    Json::Value orders(Json::arrayValue);
+
+    {
+        CoreSuspender suspend;
+
+        for (auto it = world->manager_orders.begin(); it != world->manager_orders.end(); it++)
+        {
+            Json::Value order(Json::objectValue);
+
+            order["id"] = (*it)->id;
```

Comment:
Do you need to save the IDs? This might cause issues when importing orders into different worlds.